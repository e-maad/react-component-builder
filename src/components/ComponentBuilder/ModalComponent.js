import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import ChildrenComponent from './ChildrenComponent';

const ModalComponent = ({ isOpen, width, height, children, id, appState, dispatch }) => {

    useEffect(() => {
        updateModalState(isOpen);
    }, [])

    const updateModalState = (state) => {
        dispatch({
            payload: {
                [id]: state
            },
            type: ''
        })
    }

    return <div style={{ display: appState[id] ? 'block' : 'none' }} id={id} className='modal-component modal'>
        <div style={{ width, height }} className="modal-content">
            <span>Trigger Id: {appState[`${id}Trigger`] || 'Self'}</span>
            <span className="close" onClick={() => updateModalState(false)}>&times;</span>
            {children ? <ChildrenComponent children={children} /> : ''}
        </div>
    </div>
}

const mapStateToProps = state => {
    return {
        appState: state
    };
};

export default connect(mapStateToProps)(ModalComponent);