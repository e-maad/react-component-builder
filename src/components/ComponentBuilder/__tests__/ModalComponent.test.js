import React from 'react';
import ModalComponent from '../ModalComponent';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('Modal Component', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore({});
    wrapper = shallow(
      <Provider store={store}>
        <ModalComponent isOpen={true} />
      </Provider>
    );
  });

  it('match the snapshot', () => {
    expect(renderer.create(<Provider store={store}>
      <ModalComponent isOpen={true} />
    </Provider>).toJSON()).toMatchSnapshot();
  });

  it('base element should be present', () => {
    expect(wrapper.html().includes("modal-component")).toBe(true);
  });
});