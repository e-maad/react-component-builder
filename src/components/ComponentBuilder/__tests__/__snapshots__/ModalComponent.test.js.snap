// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component match the snapshot 1`] = `
<div
  className="modal-component modal"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="modal-content"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span>
      Trigger Id: 
      Self
    </span>
    <span
      className="close"
      onClick={[Function]}
    >
      Ã—
    </span>
    
  </div>
</div>
`;
